From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ricardo Pardini <ricardo@pardini.net>
Date: Fri, 28 Nov 2025 01:36:43 +0100
Subject: net: rtl8169: define DEBUG_RTL8169

---
 drivers/net/rtl8169.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/rtl8169.c b/drivers/net/rtl8169.c
index 111111111111..222222222222 100644
--- a/drivers/net/rtl8169.c
+++ b/drivers/net/rtl8169.c
@@ -52,7 +52,7 @@
 #include <linux/delay.h>
 #include <linux/printk.h>
 
-#undef DEBUG_RTL8169
+#define DEBUG_RTL8169
 #undef DEBUG_RTL8169_TX
 #undef DEBUG_RTL8169_RX
 
-- 
Armbian

